cmd/expand-pool.go

```diff
@@ -0,0 +1,556 @@
+						objectAPI := newObjectLayerFn()
+						if objectAPI == nil {
+							continue
+						}
+						z, ok := objectAPI.(*erasureServerPools)
+						if !ok {
+							continue
+						}
+						err := z.Decommission(context.Background(), 0)
+						if err != nil {
+							log.Error("[Expand pool]: Decommission error:", err)
+							continue
+						}
+						_ = stats.saveNextStatus(false)
+					case PoolExpandStatusWaitForDecommissionComplete:
+						objectAPI := newObjectLayerFn()
+						if objectAPI == nil {
+							continue
+						}
+						z, ok := objectAPI.(*erasureServerPools)
+						if !ok {
+							continue
+						}
+						dstatus, err := z.Status(GlobalContext, 0)
+						if err != nil {
+							log.Error("[Expand pool]: Decommission Status error:", err)
+							continue
+						}
+						if dstatus.Decommission == nil || !dstatus.Decommission.Complete {
+							continue
+						}
+						_ = stats.saveNextStatus(false)
+						// rename dir right now
+						fallthrough
+					case PoolExpandStatusWaitForRenameDataDir:
+						// load pre status
+						// if not renamed, rename now
+						preStats, err := loadPoolExpandStats()
+						if err != nil {
+							continue
+						}
+						// if already renamed, skip
+						if preStats.DirHaveRenamed {
+							continue
+						}
+						err = preStats.saveNextStatus(true)
+						if err != nil {
+							continue
+						}
+						errRemote := stats.syncExpandPoolsStatusToPeer()
+						errLocal := preStats.renameDataDir()
+						// if both success, restart minio cluster
+						// or if one of them failed, manual processing is required
+						if errRemote == nil && errLocal == nil {
+							setEnvToMinioArgs("\"\"")
+							errs := globalNotificationSys.SignalService(serviceRestart)
+							for _, err := range errs {
+								if err.Err != nil {
+									if serverDebugLog {
+										log.Error("[Expand pool]: SignalService to restart error:", err.Err)
+									}
+									goto loop
+								}
+							}
+							// restart self
+							select {
+							case globalServiceSignalCh <- serviceRestart:
+							}
+							return
+						}
+						log.Errorf(`[Expand pool]: Manual processing is required here, error %v,%v
+Step1: Stop every node.
+Step2: Rename path/0 -> path/2
+Step3: Rename path/1 -> path/0")
+Step4: Remove path/2
+Step5: Restart every node. `, errRemote, errLocal)
+					}
+				case <-GlobalContext.Done():
+					return
+				}
+			}
+		}()
+		go func() {
+			ticker := time.NewTicker(time.Second * 10)
+			defer ticker.Stop()
+			for {
+				select {
+				case <-ticker.C:
+					lstat, err := os.Stat(configFile)
+					if err != nil {
+						continue
+					}
+					if !lstat.ModTime().After(ltime) {
+						continue
+					}
+					rd, err := Open(configFile)
+					if err != nil {
+						continue
+					}
+					cf := &config.ServerConfig{}
+					dec := yaml.NewDecoder(rd)
+					dec.SetStrict(true)
+					if err = dec.Decode(cf); err != nil {
+						_ = rd.Close()
+						continue
+					}
```

Comment:
why do we need this? - we only need expansion. We do not need to watch config.yaml for changes to be done automatically in this manner. 

This is too complex to support and we shouldn't do it.

Reply:
Uhmm. I can remove it. But this is no way to trigger this for now.  @harshavardhana

Reply:
Yeah but this code is dangerous we cannot have MinIO reloading config.yaml all the time

Reply:
This is still not fixed you should remove the code that re-reads config.yaml @jiuker it must be read only once during startup.

Reply:
Yeah. I am testing the feature. And will remove them if that's ok @harshavardhana